src/flamenco/runtime/fd_executor.c

```diff
@@ -913,31 +913,68 @@ dump_instr_to_protobuf( fd_exec_txn_ctx_t *txn_ctx,
   } FD_SCRATCH_SCOPE_END;
 }
```

Comment:
I think I like the general flow, but it would be nice if we had a corresponding pop() function to unroll the instr stack

Reply:
`pop` is pretty much just a stack size decrement. I was considering it but it would've just been a one-line function